Learn 274 (#1)
betrybe/flake8-linter-action: README.md

# Flake8 Linter Action

This action runs Flake8 over a project and comment on PR the errors found.

### Install dependencies

```sh
make install
# or
python3 -m pip install -r requirements.txt
```

### Run tests

```sh
make test
# or
python3 -m pytest
```

### Run flake8

```sh
make flake8
# or
python3 -m flake8 --append-config=setup.cfg
```

### Build docker image

```sh
make build
# or
docker build . -t 'pytest_evaluator_action'
```

## Evaluator Action

To call the evaluator action you must create `.github/workflows/main.yml` in the project repo with the release version.

Check the latest release [here](https://github.com/betrybe/flake8-linter-action/releases).

```yml
on:
  pull_request:
    types: [opened, synchronize]

jobs:
  evaluator_job:
    name: Evaluator Job
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - name: Flake8 Linter Step
        uses: betrybe/flake8-linter-action@v*
        id: flake8_linter
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          pr_number: ${{ github.event.number }}
```

## Inputs

### `token`

**Required**

The GitHub token to use for making API requests.

### `pr_number`

**Required**

> Pull Request number triggered the build.

Reviewer: ```suggestion
Pull Request number that triggered the build.
```

# Flake8 Linter Action

> This action run Flake8 over a project and comment on PR the errors found.

Reviewer: ```suggestion
This action runs Flake8 over a project and comments on PR the errors found.
```

### Install dependencies

```sh
make install
# or
python3 -m pip install -r requirements.txt
```

### Run tests

```sh
make test
# or
python3 -m pytest
```

### Run flake8

```sh
make flake8
# or
python3 -m flake8 --append-config=setup.cfg
```

### Build docker image

```sh
make build
# or
docker build . -t 'pytest_evaluator_action'
```

## Evaluator Action

To call the evaluator action you must create `.github/workflows/main.yml` in the project repo with the release version.

Check the latest release [here](https://github.com/betrybe/flake8-linter-action/releases).

```yml
on:
  pull_request:
    types: [opened, synchronize]

jobs:
  evaluator_job:
    name: Evaluator Job
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - name: Flake8 Linter Step
        uses: betrybe/flake8-linter-action@v*
        id: flake8_linter
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          pr_number: ${{ github.event.number }}

> 
> 

Reviewer: ```suggestion
```

```

## Inputs

### `token`

**Required**

The GitHub token to use for making API requests.

### `pr_number`

**Required**

> Pull Request number that trigger build.

Reviewer: ```suggestion
Pull Request number that triggered the build.
```

## Configure rules and analysis via `setup.cfg`

> In order to configure the ESLint analysis for your project, you must create or modify the `setup.cfg` file at the root of your project.

Reviewer: Aqui seria Flake8 né?

```suggestion
In order to configure the Flake8 analysis for your project, you must create or modify the `setup.cfg` file at the root of your project.
```

builtins =

########## OPTIONS ##########

# Report all errors, even if it is on the same line as a `# NOQA` comment.
disable-noqa = False

# Set the maximum length that any line (with some exceptions) may be.
max-line-length = 100
# Set the maximum allowed McCabe complexity value for a block of code.
max-complexity = 10
# Toggle whether pycodestyle should enforce matching the indentation of the opening bracket’s line.
# incluences E131 and E133
hang-closing = True

########## RULES ##########

# ERROR CODES
#
# E/W  - PEP8 errors/warnings (pycodestyle)
# F    - linting errors (pyflakes)
# C    - McCabe complexity error (mccabe)
#
# W503 - line break before binary operator

# Specify a list of codes to ignore.
ignore =
    W503
# Specify the list of error codes you wish Flake8 to report.
select =
    E,
    W,
    F,
    C
# Enable off-by-default extensions.
enable-extensions =

########## DOCSTRING ##########

# Enable PyFlakes syntax checking of doctests in docstrings.
doctests = True

# Specify which files are checked by PyFlakes for doctest syntax.
include-in-doctest =
# Specify which files are not to be checked by PyFlakes for doctest syntax.
exclude-in-doctest =
```

> This example was get from the this [Gist](https://gist.github.com/krnd/1f3fb6c05af365977e486c47cb7b4a72#file-flake8).

Author: ```suggestion
This example was get from this [Gist](https://gist.github.com/krnd/1f3fb6c05af365977e486c47cb7b4a72#file-flake8).
```

# flake8-linter-action

This action run Flake8 over a project and comment on PR the errors found.

### Install dependencies

```sh
make install
# or
python3 -m pip install -r requirements.txt
```

### Run tests

```sh
make test
# or
python3 -m pytest
```

### Run flake8

```sh
make flake8
# or
python3 -m flake8 --append-config=setup.cfg
```

### Build docker image

```sh
make build
# or
docker build . -t 'pytest_evaluator_action'
```

## Evaluator Action

To call the evaluator action you must create `.github/workflows/main.yml` in the project repo with the release version.

Check the latest release [here](https://github.com/betrybe/flake8-linter-action/releases).

```yml
on:
  pull_request:
    types: [opened, synchronize]

jobs:
  evaluator_job:
    name: Evaluator Job
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - name: Flake8 Linter Step
        uses: betrybe/flake8-linter-action@v*
        id: flake8_linter
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          pr_number: ${{ github.event.number }}

```

## Inputs

### `repository_main_branch`

**Required**

**Default: `master`**

> GitHub main branch to get the original tests and requirements.s

Reviewer 2: O passo de _linter_ não faz o clone dos arquivos originais. Como geralmente são arquivos de testes não é necessário adicionar essa função aqui.

Vamos ajustar a integridade de forma definitiva agora no Q4

Author: Removido
